### PR TITLE
Improve nullness annotations

### PR DESCRIPTION
**Pull request by mernst**
This pull request removes some warning suppressions, puts others in a smaller scope, and improves formatting by putting type annotations right before types instead of before method modifiers.

### CODE REVIEW

File: src/main/java/nl/tudelft/jpacman/board/Board.java [hunk 1]
```diff
@@ -30,7 +31,7 @@
 	 * Whatever happens, the squares on the board can't be null.
 	 * @return false if any square on the board is null.
 	 */
-	protected final boolean invariant() {
+	protected final boolean invariant(@UnderInitialization(Board.class) Board this) {
```
Comment (avandeursen) [inline src/main/java/nl/tudelft/jpacman/board/Board.java]: Great! I actually tried this, but did not get it to work and then gave up.

I think I did not give the `Board.class` argument to the `Underinitialization` annotation.

Comment (mernst) [inline src/main/java/nl/tudelft/jpacman/board/Board.java]: I have [improved](https://github.com/typetools/checker-framework/commit/20765ab1cc32ff0fe697a44fa753b069cd7b80d2) the documentation of `@UnderInitialization` in hopes of making it easier to use.  The improvements will appear in the online manual when we make the May 1 release of the Checker Framework.

File: src/main/java/nl/tudelft/jpacman/board/Square.java [hunk 2]
```diff
@@ -104,7 +105,7 @@ void remove(Unit occupant) {
 	 * @return <code>true</code> iff all occupants of this square have this
 	 *         square listed as the square they are currently occupying.
 	 */
-	protected final boolean invariant() {
+	protected final boolean invariant(@UnknownInitialization(Square.class) Square this) {
```
Comment (avandeursen) [inline src/main/java/nl/tudelft/jpacman/board/Square.java]: Here I originally tried `UnderInitialization`, which did not work.

The reason is of course that the `invariant` method is also called from methods other than the constructor, such as `link`. I did not think of the solution to replace `Under` by `Unknown`, but it makes sense of course.

**Comment by avandeursen** [comment]
Thanks!

Very nice indeed. Also thanks for using separate commits making it possible to revert individual changes.

The annotation/modifier order calls for a separate checkstyle / PMD rule, but I could not find an existing one.

**Comment by mernst** [comment]
To enforce correct annotation/modifier order, a tool needs to know which annotations are declaration annotations and which annotations are type annotations.
I use the scripts at `https://github.com/plume-lib/run-google-java-format` to format the entire source code, which includes the type annotations.

**Pull request merged**